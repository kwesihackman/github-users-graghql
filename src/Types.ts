export type SearchInput = {
  search: string;
};

//github search types
//https://docs.github.com/en/graphql/reference/enums#searchtype
export enum SearchType {
  USER = "USER",
  REPOSITORY = "REPOSITORY",
  ISSUE = "ISSUE",
}

export type SearchResultItemConnection = {
  after?: string;
  before?: string;
  first?: number;
  last?: number;
  query: string;
  type: SearchType;
};

export type UserEdges = {
  cursor: string;
  node: UserData;
};

export type UserData = {
  avatarUrl: string;
  login: string;
  name: string | null;
  url: string;
};
