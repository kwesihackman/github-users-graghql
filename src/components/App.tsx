import React from "react";
import Home from "./Home.Page";
function App() {
  return <Home />;
}

export default App;
