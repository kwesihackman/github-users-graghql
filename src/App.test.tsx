import React from "react";
import { MockedProvider } from "@apollo/client/testing";
import { render, queryByTestId } from "@testing-library/react";
import renderer from "react-test-renderer";
import Home from "./components/Home.Page";
import Card from "./components/Card.Component";
import { UserData } from "./Types";
import App from "./components/App";

const appSetup = () => {
  return render(
    <MockedProvider addTypename={false}>
      <App />
    </MockedProvider>
  );
};

it("Home component renders correctly", () => {
  renderer.create(<Home />);
});

it("card component renders correctly", () => {
  const user: UserData = {
    avatarUrl:
      "https://avatars1.githubusercontent.com/u/2277160?u=ec4030b31db18e0161ab276c1c446161b80e207d&v=4",
    login: "kwesihackman",
    name: "Eric Hackman",
    url: "https://github.com/kwesihackman",
  };

  renderer.create(<Card user={user} />);
});
