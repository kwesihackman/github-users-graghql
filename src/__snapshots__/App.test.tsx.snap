// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`card component renders correctly 1`] = `
<div
  className="card shadow-sm"
>
  <img
    alt="kwesihackman"
    className="card-img-top"
    src="https://avatars1.githubusercontent.com/u/2277160?u=ec4030b31db18e0161ab276c1c446161b80e207d&v=4"
  />
  <div
    className="card-body"
  >
    <h5>
      kwesihackman
    </h5>
    <div>
      Eric Hackman
    </div>
    <a
      className="btn btn-primary my-3"
      href="https://github.com/kwesihackman"
      target="_blank"
    >
      View Full Profile
    </a>
  </div>
</div>
`;
